### PR TITLE
message log sendable polymorphic field

### LINKED ISSUE #258: Add Sendable polymorphic association to MessageLog
- `sent_to` needs to be polymorphic so `MessageLog` reflects messages sent to either a Volunteer (via User) -OR- a Requester.
- I think it's ok to expose User names (via def name), but for the Requester, we prob should add a def name that's the request date and city, rather than show their name in the dropdown. 

Thank you for helping us build in sensitivity to people's privacy! Incredibly important.

### PR DESCRIPTION
**Pull request by smoline**
Resolves #258 

### Description

- changed sent_to_id to polymorphic sendable on message_log
- made Volunteer and Requester selectable options
- sendable_id dropdown field shows the Volunteer's names or the Requester's [city, state, created_at] to protect their privacy. 
- updated other message_log views and the volunteers_controller with new fields
- updated message_log model autoemail with new fields
- tests corrected with new fields

### Type of change

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not
  work as expected)

### How Has This Been Tested?

- I tested this myself in the message_log new form and it works as expected. React test suite has not been added yet for react component testing.
- Rspec tests for all tests effected by this change have been fixed and are all passing.

### CODE REVIEW

**Comment by smoline** [comment]
@maebeale this one is ready for review

File: db/migrate/20191012123313_add_sendable_to_message_logs.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class AddSendableToMessageLogs < ActiveRecord::Migration[5.2]
+  def change
+    remove_column :message_logs, :sent_to_id, :integer
```
Comment (maebeale) [inline db/migrate/20191012123313_add_sendable_to_message_logs.rb:3]: I think here we need to use remove_reference (or, keep remove_column and add remove_foreign_key) so we drop the foreign_key to send_to

**Review by maebeale — COMMENTED** [review]
Looks great, @smoline !!!

I found two other references to sent_to in the codebase. 

Would you be willing also make the following changes as part of this pr?
`message_log.rb` model, change `sent_to` in log_autoemail to:
```
...
sendable_type: person.class.to_s,
sendable_id: person.id,
...
```

And in the user factory, change the :sent_to alias to :sendable

**Comment by maebeale** [comment]
Thanks so much, @smoline !!! You're a huge boost to the project!

**Comment by maebeale** [comment]
Way to take on complex tasks -- and stick with em!

File: README.md [hunk 2]
```diff
@@ -53,7 +53,7 @@ $ rake dev:setup
 $ heroku local -f Procfile.dev
 # if you chose the local route, then you are good to go on:
   http://localhost:5000
-  $ rspec (to run the test suite)
+  $ rspec (to run the test suite) or bundle exec rspec (if the first does not work)
```
Comment (maebeale) [inline README.md:56]: p.s. best practice is to make a separate PR for a change like this so the only code in any PR is related to the issue/topic at hand, but it wasn't worth it to hold up the PR any longer this time
